crates/chainspec/src/spec.rs

```diff
@@ -154,6 +159,41 @@ impl From<ForkBaseFeeParams> for BaseFeeParamsKind {
 #[derive(Clone, Debug, PartialEq, Eq, From)]
 pub struct ForkBaseFeeParams(Vec<(Box<dyn Hardfork>, BaseFeeParams)>);
 
+/// A container for hardforks that use eip-7804 blobs.
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub struct HardforkBlobParams {
+    pub cancun: BlobParams,
+    pub prague: BlobParams,
+}
+
+impl HardforkBlobParams {
+    /// Constructs params for chainspec from a provided blob schedule.
+    /// Falls back to defaults if the schedule is empty.
+    pub fn from_schedule(blob_schedule: &BTreeMap<String, BlobScheduleItem>) -> Self {
+        let extract = |key: &str, default: fn() -> BlobParams| {
+            blob_schedule
+                .get(key)
+                .map(|item| BlobParams {
+                    target_blob_count: item.target_blob_count,
+                    max_blob_count: item.max_blob_count,
+                    ..default()
```

Comment:
this seems okay, this way we use the correct hardfork settings but use the configured target+max_blob fields